### ISSUE TITLE
Is it possible to create pvk files with OpenSSL 3.0?

### ISSUE THREAD

**Issue by ligivan** [openssl/openssl#17054]
Hello everyone.
I have a question. Is it possible to create pvk files with OpenSSL 3.0?
In previous version, it could be done with these commands:
```
OPENSSL genrsa -out test.key -passout pass:blahblah 2048
OPENSSL rsa -in test.key -inform pem -out test.pvk -outform PVK -pvk-strong -passout pass:blahblah
```

But, when i run these commands in OpenSSL 3.0, second command gives me an error:
```
300F0000:error:07880106:common libcrypto routines:ossl_pw_get_passphrase:reason(524550):crypto\passphrase.c:265:No password method specified
 300F0000:error:04800068:PEM routines:i2b_PVK:bad password read:..\crypto\pem\pvkfmt.c:1058
```

I loaded legacy provider, but unfortunately, no effect.
Do you have any suggestions or workarounds of this?

**Comment by InstanceFactory** [comment]
I have the same issue., both on Windows and Linux platform.

Did some debugging that showed me that the password provided as parameter got "lost" somewhere somehow. Unfortunately, it seems like the (or one of the) relevant code part(s) is implemented as a macro, which makes it hard to debug.

Macro is in ..\provides\implementations\encode_decode\encode_key2ms.c, line 227 "MAKE_MS_ENCODER(rsa, pvk, rsa)"

When the current encoder's encode method is called in ..\crypto\encode_decode\encode_lib.c line 635 (fkt. encode_process) "ok = current_encoder->encode(...)", the current_encoder_ctx does not contain the password, but the last para,eter &data->-ctx-pwdata does.

I've copied the pwd to the (hopefully) appropriate  area of the ctx, but it didn't help.

Would be great if someone is able to identify and fix the cause. I need these PVK files for SQL Server database encryption. "create certficates" requires a PVK file.

**Comment by t8m** [comment]
Fix in #17181

**Comment by InstanceFactory** [comment]
> Fix in #17181

@t8m : Thanks a lot for immediately fixing the issue! Downloaded the branch and was able create PVK :-)
